crates/libs/registry/src/value_iterator.rs

```diff
@@ -3,80 +3,72 @@ use super::*;
 /// An iterator of registry values.
 pub struct ValueIterator<'a> {
     key: &'a Key,
-    range: core::ops::Range<usize>,
+    idx: u32,
     name: Vec<u16>,
     data: Data,
 }
 
 impl<'a> ValueIterator<'a> {
     pub(crate) fn new(key: &'a Key) -> Result<Self> {
-        let mut count = 0;
-        let mut name_max_len = 0;
-        let mut value_max_len = 0;
-
-        let result = unsafe {
-            RegQueryInfoKeyW(
-                key.0,
-                null_mut(),
-                null_mut(),
-                null_mut(),
-                null_mut(),
-                null_mut(),
-                null_mut(),
-                &mut count,
-                &mut name_max_len,
-                &mut value_max_len,
-                null_mut(),
-                null_mut(),
-            )
-        };
-
-        win32_error(result)?;
+        let info = get_key_info(key)?;
 
         Ok(Self {
             key,
-            range: 0..count as usize,
-            name: vec![0; name_max_len as usize + 1],
-            data: Data::new(value_max_len as usize),
+            idx: 0,
+            name: vec![0; (info.value_name_max + 1) as usize],
+            data: Data::new(info.value_data_max as usize),
         })
     }
+
+    fn resize(&mut self) -> Result<()> {
+        let info = get_key_info(self.key)?;
+        self.name.resize((info.value_name_max + 1) as usize, 0);
+        self.data = Data::new(info.value_data_max as usize);
+        Ok(())
+    }
```

Comment:
Not a big deal but might be nice to default initialize and then call `resize` in the `new` method to avoid the redundant initialization here and in the other iterator.